Add position and customStyles to Dropdown component

global-usman opened the pull request:
- Added `position` prop to specify dropdown position
- Added `customStyles` prop to customize dropdown styles

global-usman committed:
Add position and customStyles to Dropdown component

- Added `position` prop to specify dropdown position
- Added `customStyles` prop to customize dropdown styles

gm-global commented on src/components/dropdown/Dropdown.tsx:
```diff
@@ -10,7 +10,8 @@ import {
 
 type DropdownType = "select" | "button";
-export type DropdownPosition = "top" | "bottom";
+export type Position = "top" | "bottom" | "left" | "right" | "center";
+export type DropdownPosition = "up" | "down";
```
Can we make these enums?

gm-global commented on src/components/dropdown/Dropdown.tsx:
```diff
@@ -21,6 +22,14 @@ export interface DropdownProps<T> {
   initialValue?: T | null;
   dropdownIcon?: boolean;
+  position: Position;
+  customStyles?: {
+    container?: CSSProperties;
+    button?: CSSProperties;
+    icon?: CSSProperties;
+    itemsWrapper?: CSSProperties;
+    item?: CSSProperties;
+  };
```
Can you define a type for `customStyles`?

gm-global commented on src/components/dropdown/__docs__/Dropdown.stories.tsx:
```diff
@@ -30,18 +30,34 @@ export const DropdownButton: Story = {
     onSelect: (option) => console.log("Selected option:", option),
     dropdownIcon: true,
+    position: "right",
+    customStyles: {
+      container: {},
+      button: {},
+      icon: {},
+      itemsWrapper: {},
+      item: {},
+    },
```
can you save `customStyles` to a const so it can be re-used in all stories

global-usman committed:
Refactor Dropdown component and Storybook stories

- Refactor `Position` and `DropdownPosition` to enums
- Introduce `CustomStylesType` for `customStyles` prop
- Update usage of `position` prop to use `Position` enum
- Define `defaultCustomStyles` object and use it in all stories
- Minor adjustments to Storybook stories
